spec/emarsys/client_spec.rb

```diff
@@ -57,6 +57,12 @@
       it 'uses 16 random bytes to generate a 32 char hex string' do
         expect(Emarsys::Client.new.header_nonce).to match(/^[0-9a-f]{32}$/i)
       end
+
+      it 'only creates the nonce once' do
```

Comment:
See above; it's already being addressed. There hasn't been a reply since Oct 14th, though, so please feel invited to lodge a separate PR that we can merge as an alternative to #15